ambry-rest/src/main/java/com.github.ambry.rest/RestServer.java

```diff
@@ -76,48 +147,42 @@
     }
+    MetricRegistry metricRegistry = clusterMap.getMetricRegistry();
     RestServerConfig restServerConfig = new RestServerConfig(verifiableProperties);
-    reporter = JmxReporter.forRegistry(clusterMap.getMetricRegistry()).build();
-    RestRequestMetricsTracker.setDefaults(clusterMap.getMetricRegistry());
+    reporter = JmxReporter.forRegistry(metricRegistry).build();
+    RestRequestMetricsTracker.setDefaults(metricRegistry);
     restServerState = new RestServerState(restServerConfig.restServerHealthCheckUri);
-    restServerMetrics = new RestServerMetrics(clusterMap.getMetricRegistry(), restServerState);
-    try {
-      RouterFactory routerFactory =
-          Utils.getObj(restServerConfig.restServerRouterFactory, verifiableProperties, clusterMap, notificationSystem);
-      router = routerFactory.getRouter();
-
-      RestResponseHandlerFactory restResponseHandlerFactory = Utils
-          .getObj(restServerConfig.restServerResponseHandlerFactory,
-              restServerConfig.restServerResponseHandlerScalingUnitCount, restServerMetrics);
-      restResponseHandler = restResponseHandlerFactory.getRestResponseHandler();
-
-      BlobStorageServiceFactory blobStorageServiceFactory = Utils
-          .getObj(restServerConfig.restServerBlobStorageServiceFactory, verifiableProperties, clusterMap,
-              restResponseHandler, router);
-      blobStorageService = blobStorageServiceFactory.getBlobStorageService();
-
-      RestRequestHandlerFactory restRequestHandlerFactory = Utils
-          .getObj(restServerConfig.restServerRequestHandlerFactory,
-              restServerConfig.restServerRequestHandlerScalingUnitCount, restServerMetrics, blobStorageService);
-      restRequestHandler = restRequestHandlerFactory.getRestRequestHandler();
-      publicAccessLogger = new PublicAccessLogger(restServerConfig.restServerPublicAccessLogRequestHeaders.split(","),
-          restServerConfig.restServerPublicAccessLogResponseHeaders.split(","));
-      NioServerFactory nioServerFactory = Utils
-          .getObj(restServerConfig.restServerNioServerFactory, verifiableProperties, clusterMap.getMetricRegistry(),
-              restRequestHandler, publicAccessLogger, restServerState);
-      nioServer = nioServerFactory.getNioServer();
-      if (router == null || restResponseHandler == null || blobStorageService == null || restRequestHandler == null
-          || nioServer == null) {
-        throw new InstantiationException("Some of the server components were null");
-      }
-    } catch (Exception e) {
-      restServerMetrics.restServerInstantiationError.inc();
-      logger.error("Exception during instantiation of RestServer", e);
-      throw new InstantiationException("Exception while creating RestServer components - " + e.getLocalizedMessage());
+    restServerMetrics = new RestServerMetrics(metricRegistry, restServerState);
+    RouterFactory routerFactory =
```

Comment:
large change because of the removal of the `try..catch` block. Actual change is only the change in arguments provided to `getObj()` for getting the request and response handler factories.